internal/scheduler/preempting_queue_scheduler.go

```diff
@@ -855,47 +851,33 @@ func (evi *Evictor) Evict(ctx *armadacontext.Context, it nodedb.NodeIterator) (*
 			return nil, err
 		}
 
+		// TODO: Should be safe to remove now.
 		for i, evictedJob := range evictedJobs {
 			if dbJob, ok := evictedJob.(*jobdb.Job); ok {
 				evictedJobs[i] = dbJob.DeepCopy()
 			}
 		}
 
 		for _, job := range evictedJobs {
-			evictedJobsById[job.GetId()] = job
+			// Create a scheduling context for when re-scheduling this job.
+			// Mark as evicted and add a node selector to ensure the job is re-scheduled onto the node it was evicted from.
+			jctx := schedulercontext.JobSchedulingContextFromJob(evi.priorityClasses, job, GangIdAndCardinalityFromAnnotations)
+
+			// TODO: This is only necessary for jobs not scheduled in this cycle.
```

Comment:
This doesn't seem like a big deal, so I would just get rid of the comment (unless you plan on addressing this in the near future).